solr/core/src/test/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/BlockPoolSlice.java

```diff
@@ -0,0 +1,1045 @@
+/*
```

Comment:
Copied straight from Hadoop code base with minor modifications to fix SecurityManager and ForkJoinPool integration. Details in SOLR-9515 and HDFS-14251